cpmdetail.py

```diff
@@ -25,41 +23,62 @@ def get_matches(puuid):
 
 
     def get_match_detail(matches, puuid):
-        matches_dict = {}
+        detailed_matches_dict = {}
         for match in matches:
             match_id = match
-            data = service.riotapi.get_match_detail(match_id, api_key)
-            data['puuid'] = puuid
-            participants = data.get("info").get("participants")
-            participant = [item for (index, item) in enumerate(participants) if item.get("puuid") == puuid]
-            participant_detail = {k: v for e in participant for (k, v) in e.items()}
-
-            match_detail = models.match_detail.MatchDetails(data)
-
-            match_participant_detail = models.match_detail.ParticipantMatchDetails(participant_detail)
-
-            match_id_dict = models.match_detail.merge(match_detail,match_participant_detail)
-
-            matches_dict[match_id] = match_id_dict
-
-        return matches_dict
-
-
-    def get_matches_ux(matches, puuid):
-        return service.riotapi.get_matches_ux(matches, puuid, api_key)
-
-
-    def get_participant(match_id, puuid):
-        # get participant id
-        return service.riotapi.get_users_participant_id(match_id, puuid, api_key)
+            match_detail_dict = service.riotapi.get_match_detail(match_id, api_key)
+            match_detail_dict['puuid'] = puuid
+            participants_dict = match_detail_dict.get("info").get("participants")
+            participant_dict = [participant for (index, participant) in enumerate(participants_dict) if
+                                participant.get("puuid") == puuid]
+            participant_detail_dict = {k: v for participant_dict in participant_dict for (k, v) in
+                                       participant_dict.items()}
+            match_detail_dict = models.match_detail.merge(models.match_detail.MatchDetails(match_detail_dict),
+                                                          models.match_detail.ParticipantMatchDetails(
+                                                              participant_detail_dict))
+            detailed_matches_dict[match_id] = match_detail_dict
+
+        return detailed_matches_dict
+
+
+    def get_index_post():
+        form = NameForm()
+        if form.validate_on_submit():
+            game_tag = form.gametag.data
+            tag_line = form.tagline.data
+            try:
+                puuid = get_puuid(game_tag, tag_line)
+            except NotFoundError as ex:
+                return render_template('index.html', form=form, message=ex.message)
+
+            matches = get_matches(puuid)
+            matches_length = len(matches)
+            if matches_length <= 0:
+                message = str(f"No account found for: {game_tag}#{tag_line}. Check for typos and try again")
+                print(message)
+                return render_template('index.html', form=form, message=message)
+
+            if matches_length > 0:
+                message = str(f"Matches for: {game_tag}#{tag_line}")
```

Comment:
I am curious how to handle different "Not found" errors in a single call stack. If the puuid isn't found that is different than if matches aren't found... I suppose the custom error needs a constructor where it can accept "what" wasn't found and add it to the message. Not your problem, just thinking out loud.

Reply:
Sounds like a good approach to research.